### ISSUE TITLE
Installing and running node on tablets.

### ISSUE THREAD

**Issue by jasonalls** [nodeschool/discussions#1374]
Hi,

I am currently researching the possibility of writing a number of standalone applications that can be easily turned into networked or internet-based applications. My avenue of exploration is the potential to develop the front end to run in a browser using HTML5, CSS3, and JavaScript 6. Communication with the backend would be via JSON requests executed by Node.js that would communicate with MongoDB using the Mongoose Database Driver.

My questions are these, can Node.js be installed on iPad, Android, and Windows Surface tablets? If so then what are the correct steps for accomplishing this? If it is not possible, then how do we make it possible?

I believe that Node.js has the potential to beat Java in the goal to write once, run anywhere. Especially since Apple have no plans to allow Java on the iPad.

The main devices that I have to support on all my future projects for which I am currently doing research for are the Apple iPad, Android Tablets, Windows Tablets, Windows Desktops and Laptops, Linux/Android Desktops and Laptops, and Apple Macintosh Desktops and Laptops.

So if I can just use HTML5, CSS3, JavaScript 6, Node.js, Mongoose, and MongoDB then I can eliminate the need to also employ the use of Java and Objective-C. Especially since my other colleagues do not want to learn and use Java or Objective-C, as they are predominantly C# and VB.NET guys.

Kind Regards,

Jason Alls
Software Developer
Lucid Research Ltd
www.lucid-research,com

**Comment by stonemirror** [comment]
On Aug 12, 2015, at 7:59 AM, jasonalls <notifications@github.com> wrote:
> 
> I am currently researching the possibility of writing a number of standalone applications that can be easily turned into networked or internet-based applications. My avenue of exploration is the potential to develop the front end to run in a browser using HTML5, CSS3, and JavaScript 6. Communication with the backend would be via JSON requests executed by Node.js that would communicate with MongoDB using the Mongoose Database Driver.
> 
> My questions are these, can Node.js be installed on iPad, Android, and Windows Surface tablets? If so then what are the correct steps for accomplishing this? If it is not possible, then how do we make it possible?
> 

Try the “Node” app on iOS.

**Comment by jasonalls** [comment]
Thank you for your quick reply. I'm having a look now at node.js for iOS on GitHub. I take it that I will have to download the project onto a MAC OS X computer running XCODE, and then compile and deploy onto the iPad. Is my understanding correct or incorrect?

**Comment by jasonalls** [comment]
By the way, I have not done any MAC development or Objective-C before. So you'll have to forgive my ignorance in this matter.

**Comment by stonemirror** [comment]
No. I’m talking about an iOS app named “Node”, available on the App Store.

**Comment by jasonalls** [comment]
Thanks for clearing that up. I shall have a look tomorrow, when I am back at work.

Date: Wed, 12 Aug 2015 09:28:37 -0700

**Comment by snuggles08** [comment]
Here's a guide and project for this (works on both Android and iOS):
* <https://github.com/jxcore/jxcore-cordova>
* <http://www.sitepoint.com/how-to-run-node-js-with-express-on-mobile-devices/>
> My questions are these, can Node.js be installed on iPad, Android, and **Windows Surface** tablets?
Only the Pro line and Surface 3 (not 2 or RT!).

**Comment by jasonalls** [comment]
Thank you for your kindness in responding to my questions stonemirror and snuggles08. The jxcore project is exactly what I am looking for. As for the database, I contacted the MongoDB developers, and have not received a definitive yes from their initial response, and I am waiting for them to get back to me via email. But I did come across another project on github that appears to be very promising, and that is called LevelDB: http://leveldb.org/. My question has been answered now, and the answer is yes I can build a full applications across multiple devices and operating system. The technology stack to do this based on your responses, and my looking into your responses is as follows: LevelDB for the database, jxcore for the webserver, and HTML5, CSS3, and JavaScript 6 for the user interface. So I am closing this issue now, and thanks again to you both. :)
